### PR TITLE
Add additional ENIs with EIPs capability

### PR DESCRIPTION
**Pull request by SweetOps**
## What

* Add additional ENIs with EIPs capability
* Add `.travis.yml` manifest

## Why

* In case we're using `terraform-aws-ec2-instance` we need option to have additional ENIs with EIPs for full NAT

### CODE REVIEW

**Comment by SweetOps** [comment]
screenshot 2017-11-10 17 47 52

**Comment by SweetOps** [comment]
tested with `additional_ips_count  = "0"` and `additional_ips_count  = "2"`

File: eni.tf [hunk 1]
```diff
@@ -0,0 +1,27 @@
+resource "aws_network_interface" "default" {
```
Comment (comeanother) [inline eni.tf]: This `aws_network_interface` is used for `additional` `aws_eip`.
Let's rename it to `additional`.

File: eni.tf [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  ]
+
+  tags {
+    Name      = "${module.label.id}"
+    Namespace = "${var.namespace}"
+    Stage     = "${var.stage}"
+  }
+}
+
+resource "aws_network_interface_attachment" "default" {
```
Comment (comeanother) [inline eni.tf]: Let's rename it to additional.

File: eni.tf [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  count                = "${var.additional_ips_count}"
+  instance_id          = "${aws_instance.default.id}"
+  network_interface_id = "${aws_network_interface.default.*.id[count.index]}"
+  device_index         = "${1 + count.index}"
+}
+
+resource "aws_eip" "additional" {
+  count             = "${var.additional_ips_count}"
```
Comment (comeanother) [inline eni.tf]: Let's keep `var.associate_public_ip_address`.

File: main.tf [hunk 4]
```diff
@@ -116,9 +116,9 @@ resource "aws_instance" "default" {
 }
 
 resource "aws_eip" "default" {
-  count    = "${var.associate_public_ip_address && var.instance_enabled ? 1 : 0}"
-  instance = "${aws_instance.default.id}"
-  vpc      = true
+  count             = "${var.associate_public_ip_address && var.instance_enabled ? 1 : 0}"
+  network_interface = "${aws_instance.default.primary_network_interface_id}"
+  vpc               = true
```
Comment (comeanother) [inline main.tf]: `"true"`

File: eni.tf [hunk 5]
```diff
@@ -0,0 +1,27 @@
+resource "aws_network_interface" "default" {
+  count     = "${var.additional_ips_count}"
+  subnet_id = "${var.subnets[0]}"
```
Comment (comeanother) [inline eni.tf]: We should, probably, drop `list` for `var.subnets`.
As `list` is not being processed now.

Comment (comeanother) [reply eni.tf]: https://cloudposse.atlassian.net/browse/CPD-115

File: eni.tf (hunk 3 above)
Comment (SweetOps) [reply eni.tf]: that's variables have using for different cases

**Comment by SweetOps** [comment]
screenshot 2017-11-13 13 39 30

**Comment by s2504s** [comment]
I've tested it. Works OK
image

File: eni.tf [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+resource "aws_network_interface" "additional" {
+  count     = "${local.additional_ips_count}"
+  subnet_id = "${var.subnet}"
+
+  security_groups = [
+    "${compact(concat(list(var.create_default_security_group ? join("", aws_security_group.default.*.id) : ""), var.security_groups))}",
+  ]
+
+  tags {
```
Comment (osterman) [inline eni.tf]: this should be `tags = "${module.label.tags}"`

File: main.tf [hunk 7]
```diff
@@ -106,7 +106,7 @@ resource "aws_instance" "default" {
 
   key_name = "${var.ssh_key_pair}"
 
-  subnet_id = "${var.subnets[0]}"
+  subnet_id = "${var.subnet}"
 
   tags {
```
Comment (osterman) [inline main.tf]: this should be `tags = "${module.label.tags}"`

File: eni.tf (hunk 6 above)
Comment (SweetOps) [reply eni.tf]: will be fixed in https://github.com/cloudposse/terraform-aws-ec2-instance/pull/15

File: main.tf (hunk 7 above)
Comment (SweetOps) [reply main.tf]: will be fixed in https://github.com/cloudposse/terraform-aws-ec2-instance/pull/15

**Pull request merged**